src/Sylius/Bundle/AdminBundle/Resources/config/routing/product.yml

```diff
@@ -0,0 +1,52 @@
+sylius_admin_product:
+    resource: |
+        alias: sylius.product
+        section: admin
+        templates: SyliusAdminBundle:Crud
+        except: ['show']
+        redirect: update
+        grid: sylius_admin_product
+        vars:
+            all:
+                subheader: sylius.ui.manage_your_products
+                templates:
+                    form: SyliusAdminBundle:Product/Form:_form.html.twig
+            index:
+                icon: diamond
+    type: sylius.resource
+
+sylius_admin_product_create:
+    path: /products/new
+    methods: [GET, POST]
+    defaults:
+        _controller: sylius.controller.product:createAction
+        _sylius:
+            section: admin
```

Comment:
I had to add it here, because one of twig templates would look for `sylius_product_create` instead of `sylius_admin_product_create`